### PR TITLE
Add dbengine RAM usage statistics

### LINKED ISSUE #7020: Add memory usage statistics for dbengine
##### Feature idea summary
Add memory usage statistics for dbengine instances.

### PR DESCRIPTION
**Pull request by mfundul**
##### Summary
Fixes #7020
##### Component Name
database/engine
##### Additional Information
Add dbengine RAM usage statistics.

### CODE REVIEW

File: daemon/global_statistics.c [hunk 1]
```diff
@@ -818,6 +818,63 @@ void global_statistics_charts(void) {
...
+            static RRDDIM *rd_cached = NULL;
+            static RRDDIM *rd_pinned = NULL;
+            static RRDDIM *rd_metadata = NULL;
+
+            collected_number cached_pages, pinned_pages, API_producers, populated_pages, metadata, pages_on_disk,
+            page_cache_descriptors;
+
+            if (unlikely(!st_ram_usage)) {
+                st_ram_usage = rrdset_create_localhost(
+                        "netdata"
```
Comment (thiagoftsm) [inline daemon/global_statistics.c]: This is not aligned with the others.

**Comment by mfundul** [comment]
It shouldn't be like that. Try refreshing your browser.

**Comment by thiagoftsm** [comment]
> It shouldn't be like that. Try refreshing your browser.

I refresh it and I will let it running during few hours.

**Comment by mfundul** [comment]
> 
> 
> Hi @mfundul ,
> 
> I verified a behavior completely unexpected here in the chart that you are bringing, on the right side we have `collectors = 6.8` and `metadata = 41.3`, but the collectors are bigger than the metdata on the left side. Does this strange behavior has relationship with this PR? Or are we in front a bug in front end?
> 
> strange_behavior

The blue part is the metadata and is bigger than the metadata on the left side. I don't quite understand.

**Comment by thiagoftsm** [comment]
> The blue part is the metadata and is bigger than the collectors on the left side. I don't quite understand.

I was hoping  to see a bigger blue rectangle instead a bigger red, due their values.

**Comment by mfundul** [comment]
> 
> 
> > The blue part is the metadata and is bigger than the collectors on the left side. I don't quite understand.
> 
> I was hoping to see a bigger blue rectangle instead a bigger red, due their values.

But the blue is bigger.

**Comment by thiagoftsm** [comment]
> > > The blue part is the metadata and is bigger than the collectors on the left side. I don't quite understand.
> > 
> > 
> > I was hoping to see a bigger blue rectangle instead a bigger red, due their values.
> 
> But the blue is bigger.

Yes, but why is the red on top of it when it has a small value? This is what is strange.

**Comment by mfundul** [comment]
> 
> 
> > > > The blue part is the metadata and is bigger than the collectors on the left side. I don't quite understand.
> > > 
> > > 
> > > I was hoping to see a bigger blue rectangle instead a bigger red, due their values.
> > 
> > 
> > But the blue is bigger.
> 
> Yes, but why is the red on top of it when it has a small value? This is what is strange.

It's a stacked chart similar to `system.ram`, the order always remains the same.

**Comment by thiagoftsm** [comment]
> It's a stacked chart similar to `system.ram`, the order always remains the same.

I understood now, but I continue find strange , I will approve it now.

**Pull request merged**